source/1_engine/rex_engine/include/rex_engine/string/stringid.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "rex_engine/string/stringentry.h"
+#include "rex_engine/types.h"
+#include "rex_std/ostream.h"
+#include "rex_std/string.h"
+
+namespace rex
+{
+  class StringID
+  {
+  public:
+    StringID();
+    explicit StringID(rsl::string_view characters);
+
+    explicit operator u32() const;
+
+    bool operator==(const StringID& other) const;
+    bool operator!=(const StringID& other) const;
+
+    bool operator==(const StringEntryID& entryID) const;
+    bool operator!=(const StringEntryID& entryID) const;
+
+    rsl::string_view to_string_view() const;
+    bool is_none() const;
+    u32 value() const;
+
+  private:
+    /** Index into the StringID array (used to find String portion of the string/number pair used for comparison) */
+    StringEntryID m_comparison_index;
+  };
+
+  StringID create_sid(rsl::string_view stringView);
+
+  bool operator==(rsl::string_view s, const StringID& sid);
```

Comment:
I have somewhat of a coding convention that everything which is a class or struct I pass by ref, I can pass it by value if you want. The reason why this is is because if the class/struct changes I do not have to worry I am making expensive copies.

Reply:
there's a class in rsl that makes it a reference if it's bigger than 1 pointer size, a non-ref if not if you're interested

Reply:
Which is?

Reply:
rsl::call_traits<T>::param_type
a bit verbose, I know